### ISSUE TITLE
Improve withOptions qualifier detection in GenericDefinitionWithoutQualifier

### ISSUE THREAD

**Issue by kirich1409** [androidbroadcast/Koin-Detekt#52]
## Context
Code review comments from PR #32: https://github.com/androidbroadcast/Koin-Detekt/pull/32

### 1. False negative with withOptions blocks
The rule checks for qualifiers in `valueArguments` but does not check for `withOptions` blocks that may contain `qualifier = named()` assignments.

Example that is incorrectly flagged:
```kotlin
single { listOf<String>() } withOptions { qualifier = named("strings") }
```

This has a qualifier, but the rule doesn't detect it.

### 2. Nested control flow not detected
The rule only checks immediate child statements via `bodyExpression.statements`. If the withOptions block contains nested control flow (e.g., `if` statements or `when` expressions) with qualifier assignments inside, they won't be detected.

Example:
```kotlin
withOptions { if (condition) qualifier = named("x") }
```

## Proposed Solution

1. Check for `withOptions` chained to the definition using PSI traversal
2. Look for `qualifier = named(...)` or `qualifier = qualifier(...)` in the withOptions lambda body
3. For nested control flow, consider traversing deeper into the PSI tree or documenting the limitation

```kotlin
// Check for qualifier specified in a chained withOptions { qualifier = named("...") } block
val hasQualifierInOptions = (expression.parent as? KtDotQualifiedExpression)
    ?.selectorExpression
    ?.let { it as? KtCallExpression }
    ?.takeIf { it.calleeExpression?.text == "withOptions" }
    ?.lambdaArguments
    ?.firstOrNull()
    ?.getLambdaExpression()
    ?.bodyExpression
    ?.text
    ?.let { bodyText ->
        bodyText.contains("qualifier =") &&
            (bodyText.contains("named(") || bodyText.contains("qualifier("))
    } ?: false

val hasQualifier = hasQualifierInArguments || hasQualifierInOptions
```

## Priority
IMPORTANT - Can cause false positives on valid code with qualifiers.

## Related
- PR #32: https://github.com/androidbroadcast/Koin-Detekt/pull/32
